src/main/java/site/coach_coach/coach_coach_server/coach/service/CoachService.java

```diff
@@ -90,72 +88,94 @@ public CoachDetailDto getCoachDetail(User user, Long coachId) {
-			}
+		List<Long> sportsList = (sports != null && !sports.isEmpty()) ? parseSports(sports) : allSportsIds;
+		sportsList = getExistingSportsList(sportsList);
+
+		if (sportsList.isEmpty() && (sports != null && !sports.isEmpty())) {
+			return new CoachListResponse(List.of(), 0, page);
 		}
 
-		Page<Coach> coachesPage;
-		if (review != null && review) {
-			coachesPage = coachRepository.findAllWithReviewsSorted(sportsList, search, pageable);
-		} else if (liked != null && liked) {
-			coachesPage = coachRepository.findAllWithLikesSorted(sportsList, search, pageable);
-		} else if (latest != null && latest) {
-			coachesPage = coachRepository.findAllWithLatestSorted(sportsList, search, pageable);
-		} else if (my != null && my) {
-			coachesPage = coachRepository.findMyCoaches(user.getUserId(), sportsList, search, pageable);
-		} else {
-			throw new InvalidQueryParameterException(ErrorMessage.INVALID_QUERY_PARAMETER);
+		Page<Coach> coachesPage = fetchCoachesPage(user, sportsList, search, pageable, review, liked, latest, my);
+
+		if (coachesPage.isEmpty() && page == 1) {
+			return new CoachListResponse(List.of(), 0, page);
 		}
 
 		if (page > coachesPage.getTotalPages()) {
 			throw new NotFoundPageException(ErrorMessage.NOT_FOUND_PAGE);
 		}
 
 		List<CoachListDto> coaches = coachesPage.stream()
-			.map(coach -> {
-				List<Review> reviews = reviewRepository.findByCoach_CoachId(coach.getCoachId());
-				double averageRating = reviews.stream().mapToInt(Review::getStars).average().orElse(0.0);
-				int countOfReviews = reviews.size();
-
-				boolean isLiked = isLikedByUser(user, coach);
-				int countOfLikes = getCountOfLikes(coach);
-
-				List<CoachingSportDto> coachingSports = coach.getCoachingSports().stream()
-					.map(cs -> new CoachingSportDto(
-						cs.getSport().getSportId(),
-						cs.getSport().getSportName()
-					))
-					.collect(Collectors.toList());
-				return CoachListDto.builder()
-					.coachId(coach.getCoachId())
-					.coachName(coach.getUser().getNickname())
-					.localAddress(coach.getUser().getLocalAddress())
-					.profileImageUrl(coach.getUser().getProfileImageUrl())
-					.coachIntroduction(coach.getCoachIntroduction())
-					.coachingSports(coachingSports)
-					.countOfReviews(countOfReviews)
-					.reviewRating(averageRating)
-					.isLiked(isLiked)
-					.countOfLikes(countOfLikes)
-					.build();
-			})
+			.map(coach -> getCoachListDto(coach, user))
 			.collect(Collectors.toList());
 
 		return new CoachListResponse(coaches, (int)coachesPage.getTotalElements(), page);
 	}
 
+	private List<Long> getExistingSportsList(List<Long> sportsList) {
+		if (sportsList == null || sportsList.isEmpty()) {
+			return List.of();
+		}
+
+		List<CoachingSport> existingSports = coachingSportRepository.findAllBySport_SportIdIn(sportsList);
+		List<Long> validSportIds = existingSports.stream()
+			.map(cs -> cs.getSport().getSportId())
+			.collect(Collectors.toList());
+
+		return validSportIds;
```

Comment:
바로 return 처리할 수도 있을 것 같아용!

```suggestion
		return existingSports.stream()
			.map(cs -> cs.getSport().getSportId())
			.collect(Collectors.toList());
```